packages/core/src/data-grid/data-grid.tsx

```diff
@@ -1313,23 +1325,75 @@ const DataGrid: React.ForwardRefRenderFunction<DataGridRef, DataGridProps> = (p,
     );
 
     const stickyShadow = React.useMemo(() => {
```

Comment:
This memo now serves very little purpose. It will re-evaluate almost ever scroll step. If you compute `opacityX` and `opacityY` outside the memo and then pass those in it will only reevaluate when needed.